ipfshttpclient/filescanner.py

```diff
@@ -572,29 +599,66 @@ def close(self) -> None:
 			pass
 	
 	@staticmethod
-	def _join_dirs_and_files(dirnames: ty.List[ty.AnyStr], filenames: ty.List[ty.AnyStr]) \
-	    -> ty.Iterator[ty.Tuple[ty.AnyStr, bool]]:
+	def _join_dirs_and_files(dirnames: ty.List[AnyStr], filenames: ty.List[AnyStr]) \
+	    -> ty.Iterator[ty.Tuple[AnyStr, bool]]:
 		for dirname in dirnames:
-			yield (dirname, True)
+			yield dirname, True
 		
 		for filename in filenames:
-			yield (filename, False)
-	
+			yield filename, False
+
+	@staticmethod
+	def _walk_separator(
+			matcher: Matcher[AnyStr],
+			directory_str: ty.Optional[AnyStr]
+	) -> ty.Union[bytes, str]:
+		"""
+		Determine which separator to use.
+
+		Because os.fsencode can return a byte array, we must allow returning a byte array,
+		regardless of AnyType.
+		"""
+
+		if directory_str is not None:
+			return utils.maybe_fsencode(os.path.sep, directory_str)
+		elif matcher is not None and matcher.is_binary:
+			return os.fsencode(os.path.sep)
+		else:
+			return os.path.sep
+
+	@staticmethod
+	def _walk_wide(
+			dot: AnyStr,
+			directory: ty.Union[AnyStr, int],
+			follow_symlinks: bool
+	) -> ty.Iterator[ty.Tuple[AnyStr, ty.List[AnyStr], ty.List[AnyStr], ty.Optional[int]]]:
+		"""
+		Return a four-part tuple just like os.fwalk does, even if we won't use os.fwalk.
+
+		The directory file descriptor will be None when os.fwalk is not used.
+		"""
+
+		if isinstance(directory, int):
+			yield from os.fwalk(dot, dir_fd=directory, follow_symlinks=follow_symlinks)
+		else:
+			for dir_path, dir_names, file_names in os.walk(directory, followlinks=follow_symlinks):
+				yield dir_path, dir_names, file_names, None
+
 	def _walk(
 			self,
-			directory: ty.Union[ty.AnyStr, int],
-			directory_str: ty.Optional[ty.AnyStr],
-			matcher: Matcher[ty.AnyStr],
+			directory: ty.Union[AnyStr, int],
+			directory_str: ty.Optional[AnyStr],
+			matcher: Matcher[AnyStr],
 			follow_symlinks: bool,
 			intermediate_dirs: bool
-	) -> ty.Generator[FSNodeEntry, ty.Any, None]:
-		if directory_str is not None:
-			sep = utils.maybe_fsencode(os.path.sep, directory_str)
-		elif matcher is not None and matcher.is_binary:
-			sep = os.fsencode(os.path.sep)  # type: ignore[assignment]
-		else:
-			sep = os.path.sep  # type: ignore[assignment]
-		dot = utils.maybe_fsencode(".", sep)  # type: ty.AnyStr
+	) -> ty.Generator[FSNodeEntry[AnyStr], None, None]:
+		separator = self._walk_separator(matcher=matcher, directory_str=directory_str)
+
+		# TODO: Because os.fsencode can return a byte array, we need to refactor how we use 'sep'
+		# noinspection PyTypeHints
```

Comment:
It's a `PyCharm` IDE warning suppression at the statement level.

Reply:
Why does it need this when we already have a `type: ignore` comment below?

If the answer is “because PyCharm doesn't understand that” then please remove this comment anyways. It is not unreasonable for PyCharm using people to just ignore the warning after looking at that line and I want to keep PyCharm-specific changes to an absolute minimum. Not everyone can and wants to use PyCharm and so they cannot use PyCharm's type checker and (like me) won't know what that comment is about, OTOH any Python dev can easily access mypy thanks to `tox` and the shipped config so that's what is primarily used here.

Reply:
I'll remove it. The scope of this issue was a single line.

In some sense, it's hiding a problem that should be flagged anyways.